internal/pkg/aws/ec2/ec2.go

```diff
@@ -20,35 +20,6 @@ const (
 	TagFilterName = "tag:%s"
 )
 
-// ListVPCSubnetsOpts sets up optional parameters for ListVPCSubnets function.
-type ListVPCSubnetsOpts func([]*ec2.Subnet) []*ec2.Subnet
-
-// FilterForPublicSubnets is used to filter to get public subnets.
-func FilterForPublicSubnets() ListVPCSubnetsOpts {
```

Comment:
Should we not be figuring out if a subnet has an internet gateway? so that we filter out the ones that are not relevant

Reply:
It would be very difficult to tell since we need to tell
1. if the VPC has an internet gateway attached to it
2. describe the route table check if it has route to the internet gateway and check if the route table is associated with the subnet (we would need to do it for every subnet within the VPC)

Reply:
Is the drawback that this is too fragile? If we can provide a good experience by making these API calls and if there are no valid subnets, prompt customers what changes they'd need to make to their subnet in order for it to be useful to Copilot?

Reply:
It's not too bad I think. We can describe the route tables for a given VPC from the user:
```
$ aws ec2 describe-route-tables --filters Name=vpc-id,Values=vpc-03e4614a3d0217256
```
In the response we'll look for `SubnetId` under `Associations` and for `Routes` to have a `GatewayId` that starts with `igw-`:
<details>
  <summary> View JSON response </summary>
  
  ```json
"Associations": [
    {
        "Main": false,
        "RouteTableAssociationId": "rtbassoc-096f8559b5e31d9d3",
        "RouteTableId": "rtb-073d682766628c37f",
        "SubnetId": "subnet-018ccb78d353cec9b",
        "AssociationState": {
            "State": "associated"
        }
    },
    {
        "Main": false,
        "RouteTableAssociationId": "rtbassoc-0ebf8e431225c1b20",
        "RouteTableId": "rtb-073d682766628c37f",
        "SubnetId": "subnet-005dc83a6473feee7",
        "AssociationState": {
            "State": "associated"
        }
    }
],
"PropagatingVgws": [],
"RouteTableId": "rtb-073d682766628c37f",
"Routes": [
    {
        "DestinationCidrBlock": "10.0.0.0/16",
        "GatewayId": "local",
        "Origin": "CreateRouteTable",
        "State": "active"
    },
    {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": "igw-058421418aeacb2f0",
        "Origin": "CreateRoute",
        "State": "active"
    }
]
```
</details>

The only downside is we won't have tags associated with the subnets, but we can make two network calls to get that info.

Reply:
ok addressed!